G4_DailyReport/src/main/java/com/example/G4_DailyReport/service/DepartmentService.java

```diff
@@ -58,4 +61,42 @@ public void deleteById(UUID id) {
     public Page<User> findManagers(UUID id, Pageable pageable) {
         return userRepository.findAllByDepartmentIdAndPositionName(id, "Manager", pageable);
     }
+
+    @Transactional(propagation = Propagation.REQUIRED, rollbackFor = Exception.class)
+    public void addManager(UUID departmentId, UUID managerId) {
+        User manager = userRepository.findById(managerId).orElse(null);
+        Department department = departmentRepository.findById(departmentId).orElse(null);
+        if(manager == null || department == null) {
+            throw new DataIntegrityViolationException("Manager or Department not found");
+        }
+        if(manager.getDepartment() != null) {
+            throw new DataIntegrityViolationException("Manager already has a department");
+        }
+        manager.setDepartment(department);
+        userRepository.save(manager);
+    }
```

Comment:
Chị thấy cái method này là 1 phần dc tách ra trong method `addManagers` 
thì method này `addManager` nên là private, và ko cần dùng `@Transactional(propagation = Propagation.REQUIRED, rollbackFor = Exception.class)` nhỉ
```suggestion
    private void addManager(UUID departmentId, UUID managerId) {
        User manager = userRepository.findById(managerId).orElse(null);
        Department department = departmentRepository.findById(departmentId).orElse(null);
        if(manager == null || department == null) {
            throw new DataIntegrityViolationException("Manager or Department not found");
        }
        if(manager.getDepartment() != null) {
            throw new DataIntegrityViolationException("Manager already has a department");
        }
        manager.setDepartment(department);
        userRepository.save(manager);
    }
```